File: aoc2016/aoc2016-groovy/src/test/java/de/havox_design/aoc2016/day21/Day21Test.java
package de.havox_design.aoc2016.day21;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

class Day21Test {

    @ParameterizedTest
    @MethodSource("getDataForPart1")
    void testPart1(String fileName, String expected, String input) {
        Assertions.assertEquals(expected, ScrambledLettersAndHash.solvePart1(fileName, input));
    }

    private static Stream<Arguments> getDataForPart1() {
        return Stream.of(
                Arguments.of("de/havox_design/aoc2016/day21/day21Sample.txt", "decab", "abcde")
        );
    }
}

File: aoc2023/aoc2023-kotlin/src/test/kotlin/de/havox_design/aoc2023/day21/Day21Test.kt
package de.havox_design.aoc2023.day21

import de.havox_design.aoc.utils.kotlin.helpers.tests.shouldBe
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource
import java.util.stream.Stream

class Day21Test {
    @ParameterizedTest
    @MethodSource("getDataForTestSolvePart1")
    fun testSolvePart1(filename: String, expectedResult: Long) =
        StepCounter(filename).solvePart1(6).shouldBe(expectedResult)

    companion object {
        @JvmStatic
        private fun getDataForTestSolvePart1(): Stream<Arguments> =
            Stream.of(
                Arguments.of("de/havox_design/aoc2023/day21/day21Sample.txt", 16L)
            )
    }
}
